app/views/play/level/tome/SpellView.coffee

```diff
@@ -988,6 +993,37 @@ module.exports = class SpellView extends CocoView
     @spell.transpile()  # TODO: is there any way we can avoid doing this if it hasn't changed? Causes a slight hang.
     @updateAether false, false
 
+  onWebDevError: (error) ->
+    annotations = @aceSession.getAnnotations()
+    offsetError = _.merge {}, error, { line: error.line + @linesBeforeScript(@getSource()) }
+    problem = new Problem({ error: offsetError, @ace, levelID: @options.levelID })
+    if _.any(@problems, (preexistingProblem) -> problem.isEqual(preexistingProblem))
+      problem.destroy()
+      return
+
+    @problems.push problem
+
+    lineOffsetPx = 0
+    for i in [0...problem.row]
+      lineOffsetPx += @aceSession.getRowLength(i) * @ace.renderer.lineHeight
+    lineOffsetPx -= @ace.session.getScrollTop()
+    Backbone.Mediator.publish 'tome:show-problem-alert', problem: problem, lineOffsetPx: Math.max lineOffsetPx, 0
+
+    # @saveUserCodeProblem(aether, aetherProblem) # TODO: Enable saving of web-dev user code problems
+    annotations.push problem.annotation if problem.annotation
+    @reallySetAnnotations annotations
+    Backbone.Mediator.publish 'tome:problems-updated', spell: @spell, problems: @problems, isCast: true
+
+    # TODO: Unset this on rerun
+    @ace.setStyle 'user-code-problem'
+    # TODO: When is this unset?
+    @ace.setStyle 'spell-cast'
+
+  linesBeforeScript: (html) ->
+    # TODO: refactor, make it work with multiple scripts. What to do when error is in level-creator's code?
+    _.size(html.split('<script>')[0].match(/\n/g))
```

Comment:
We have a concept of user-code problems and non-user-code problems. User code problems are recoverable and should just show an error message, no problem. Non-user-code problems we bend over backwards to make sure we never have and we report them to level design team via email and get them fixed, sometimes with aggressive argument validation and things like this. So at some point if we are seeing errors in level creator's code, we'll need to make sure we can trigger non-user-code problems properly so they are reported.